src/Documentation/Event-Sourcing/LogConsistencyProviders.md

```diff
@@ -0,0 +1,54 @@
+---
+layout: page
+title: Included Log-Consistency Providers
+---
+
+# Built-In Log-Consistency Providers
+
+The `Microsoft.Orleans.EventSourcing` package includes several log-consistency providers that cover basic scenarios suitable to get started, and allow some extensibility.
+
+
+### Orleans.EventSourcing.**StateStorage**.LogConsistencyProvider
+
+This provider stores *grain state snapshots*, using a standard storage provider that can be configured independently. 
+
+The data that is kept in storage is an object that contains both the grain state (as specified by the first type parameter to `JournaledGrain`) and some meta-data (the version number, amd a special tag that is used to avoid duplication of events when storage accesses fail).
+
+Since the entire grain state is read/written every time we access storage, this provider is not suitable for objects whose grain state is very large. 
+
+This provider does not support `RetrieveConfirmedEvents`: it cannot retrieve the events from storage because the events are not persisted.
+
+### Orleans.EventSourcing.**LogStorage**.LogConsistencyProvider
+
+This provider stores *the complete event sequence as a single object*, using a standard storage provider that can be configured independently.
+
+The data that is kept in storage is an object that contains a `List<EventType> object`, and some meta-data (a special tag that is used to avoid duplication of events when storage accesses fail).
+
+This provider does support `RetrieveConfirmedEvents`. All events are always available and kept in memory.
+
+Since the whole event sequence is read/written every time we access storage, this provider is _not suitable for use in production_, unless the event sequences are guaranteed to remain pretty short. The main purpose of this provider is to illustrate the semantics of the event sourcing, and for samples/testing environments.
```

Comment:
I'm not sure it will be really efficient... The current storage provider save and read the whole state from storage every time. Event Stores read/write each record... By definition they are diff... Idk how you plan to get it working. Think that you can have hundreds of events per entity...